k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -82,11 +88,11 @@ public long getSize() {
         throw new IllegalStateException("Data must be fully written before it can be read!");
     }
 
-    public File getFile() throws IOException {
+    public Uri getProviderUri(String mimeType) throws IOException {
```

Comment:
`AttachmentInfoExtractor` seems to be a better place for this kind of information. It already knows about different Part/Body types. And there's currently exactly one implementation of FileProviderInterface. So moving the code there should be straightforward. This would be similar to what we're doing with `LocalPart` and `AttachmentProvider`.

`AttachmentInfoExtractor` is also closer to the layer that should have this information. Putting the information (or a reference to it) into the Body object itself feels wrong.